rlclientlib/multistep.cc

```diff
@@ -2,27 +2,27 @@
 #include "err_constants.h"
 
 namespace reinforcement_learning {
-  void episode_history::update(const char* event_id, const char* previous_event_id, string_view context, const ranking_response& resp) {
-    _depths[event_id] = this->get_depth(previous_event_id) + 1;
+  void episode_history::update(string_view event_id, string_view previous_event_id, string_view context, const ranking_response& resp) {
+    _depths[std::string(event_id)] = this->get_depth(previous_event_id) + 1;
```

Comment:
This is not great... We get heterogeneous lookup in c++14 for map I believe. Doesn't help us here though